Server/VikingVault.Services/TransactionService.cs

```diff
@@ -47,41 +47,50 @@ public void AddTransactionsForTransferFunds(TransferFundsModel transferData)
 
                 Transaction senderTransaction = new Transaction
                 {
-                    User = transferData.Sender,
+                    Sender = transferData.Sender,
                     Type = "Transfer",
                     Currency = transferData.Currency,
                     Date = DateTime.Now,
                     Amount = -transferData.AmountSent,
-                    OtherParty = reciever.FirstName + reciever.LastName
-                };
-
-                Transaction recieverTransaction = new Transaction
-                {
-                    User = reciever,
-                    Type = "Transfer",
-                    Currency = transferData.Currency,
-                    Date = DateTime.Now,
-                    Amount = transferData.AmountSent,
-                    OtherParty = transferData.Sender.FirstName + transferData.Sender.LastName
+                    Receiver = reciever,
+                    Details = transferData.TransferDetails
                 };
 
                 AddTransaction(senderTransaction);
-                AddTransaction(recieverTransaction);
             }
             catch
             {
                 throw new DatabaseException();
             }
         }
 
-        public List<Transaction> GetTransactions(string userId)
+        public List<TransactionDTO> GetTransactions(string userId)
         {
             int uid = int.Parse(userId);
-            return _dbContext.Transactions
-                .Where(t => t.User.Id == uid)
+            var transactions =  _dbContext.Transactions
+                .Include(t => t.Sender)
+                .Include(t => t.Receiver)
+                .Where(t => t.Sender.Id == uid || t.Receiver.Id == uid)
                 .OrderByDescending(t => t.Date)
                 .Take(5)
```

Comment:
This `.Take()` amount should either be passed as an argument to the function,or extracted into a constant or you make two overloaded methods, one which gets the with the default amount, one where you pass in the amount.